apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs

```diff
@@ -989,7 +989,128 @@ public async Task CanExecuteReadPartitionedReadCommand()
+            yield return new object[] { "DecimalToNumeric", 3.14m, SpannerDbType.Numeric };
+            yield return new object[] { "DecimalToPgNumeric", 3.14m, SpannerDbType.PgNumeric };
+        }
+         
+        [Theory]
+        [MemberData(nameof(ConfiguredSpannerDbTypes))]
+        public void ExecuteStreamingSql_ParameterDefaultsToConfiguredType(string clrToSpannerTypeMapping, object value, SpannerDbType spannerDbType)
+        {
+            var builder = new SpannerConnectionStringBuilder();
+            if (!string.IsNullOrWhiteSpace(clrToSpannerTypeMapping))
+            {
+                builder.ClrToSpannerTypeDefaultMappings = clrToSpannerTypeMapping;
+            }
+
+            var parameter = new SpannerParameter
+            {
+                ParameterName = "p",
+                Value = value
+            };
+            var actualValue = RunExecuteStreamingSqlWithParameter(builder, parameter).Fields["p"];
+            var expectedValue = spannerDbType.ToProtobufValue(parameter.Value);
+            Assert.Equal(expectedValue, actualValue);
+        }
+
+        public static IEnumerable<object[]> ConfiguredClrTypes()
+        {
+            // Format : SpannerToClrTypeDefaultMappings value, value, SpannerDbType, expected CLR type.
+            // Float64 mappings.
+            yield return new object[] { default, 3.14d, SpannerDbType.Float64, typeof(double) };
+            yield return new object[] { "Float64ToFloat", 3.14d, SpannerDbType.Float64, typeof(float) };
+            yield return new object[] { "Float64ToDouble", 3.14d, SpannerDbType.Float64, typeof(double) };
+            yield return new object[] { "Float64ToDecimal", 3.14d, SpannerDbType.Float64, typeof(decimal) };
+            yield return new object[] { "Float64ToSpannerNumeric", 3.14d, SpannerDbType.Float64, typeof(SpannerNumeric) };
+            yield return new object[] { "Float64ToPgNumeric", 3.14d, SpannerDbType.Float64, typeof(PgNumeric) };
+        }
+
+        [Theory]
+        [MemberData(nameof(ConfiguredClrTypes))]
+        public void ExecuteReadRequest_ColumnDefaultsToConfiguredClrType(string spannerToClrTypeMapping, object dbValue, SpannerDbType dbType, System.Type clrType)
+        {
+            var builder = new SpannerConnectionStringBuilder();
+            if (!string.IsNullOrWhiteSpace(spannerToClrTypeMapping))
+            {
+                builder.SpannerToClrTypeDefaultMappings = spannerToClrTypeMapping;
+            }
+            var options = SpannerConversionOptions.ForConnectionStringBuilder(builder);
+            Value protobufValue = RunReadRequest(dbType, dbValue).Values[0];
+            // This is a hack, but simpler to do what DataReader does behind the scenes.
+            var clrValue = dbType.ConvertToClrType<object>(protobufValue, options);
+            // Check that CLR type of value is as expected.
+            Assert.Equal(clrType, clrValue.GetType());
+            // Check the values.
```

Comment:
This only checks the value for double. To be honest, I think it's probably simpler to just remove this bit and assume that if we've got the right type, the right conversion has taken place - that's tested elsewhere already.